compiler/rustc_pattern_analysis/src/lib.rs

```diff
@@ -87,11 +78,9 @@ pub trait TypeCx: Sized + fmt::Debug {
 /// Context that provides information global to a match.
 #[derive(derivative::Derivative)]
 #[derivative(Clone(bound = ""), Copy(bound = ""))]
-pub struct MatchCtxt<'a, 'p, Cx: TypeCx> {
+pub struct MatchCtxt<'a, Cx: TypeCx> {
```

Comment:
Can you replace `MatchCtxt<'a, Cx>` with `&'a Cx` in all of its usages?

Reply:
No, there are other things I'll need to put in there later

Reply:
Like what?

Reply:
Currently planning:
- the `overlapping_range_endpoints` Vec I introduce in https://github.com/rust-lang/rust/pull/119396
- some kind of step counter to gracefully abort on exponential cases (also requested by r-a https://github.com/rust-lang/rust-analyzer/issues/9528)